test/0.8.25/Protocol__Deployment.t.sol

```diff
@@ -0,0 +1,302 @@
+
+    function burnExternalShares(uint256 _amountOfShares) external;
+
+    function setMaxExternalRatioBP(uint256 _maxExternalRatioBP) external;
+
+    function initialize(address _lidoLocator, address _eip712StETH) external payable;
+
+    function resumeStaking() external;
+
+    function resume() external;
+
+    function setStakingLimit(uint256 _maxStakeLimit, uint256 _stakeLimitIncreasePerBlock) external;
+}
+
+interface IKernel {
+    function acl() external view returns (IACL);
+
+    function newAppInstance(
+        bytes32 _appId,
+        address _appBase,
+        bytes calldata _initializePayload,
+        bool _setDefault
+    ) external;
+}
+
+interface IACL {
+    function initialize(address _permissionsCreator) external;
+
+    function createPermission(address _entity, address _app, bytes32 _role, address _manager) external;
+
+    function hasPermission(address _who, address _where, bytes32 _what) external view returns (bool);
+}
+
+interface IDaoFactory {
+    function newDAO(address _root) external returns (IKernel);
+}
+
+struct LidoLocatorConfig {
+    address accountingOracle;
+    address depositSecurityModule;
+    address elRewardsVault;
+    address legacyOracle;
+    address lido;
+    address oracleReportSanityChecker;
+    address postTokenRebaseReceiver;
+    address burner;
+    address stakingRouter;
+    address treasury;
+    address validatorsExitBusOracle;
+    address withdrawalQueue;
+    address withdrawalVault;
+    address oracleDaemonConfig;
+    address accounting;
+    address wstETH;
+}
+
+contract BaseProtocolTest is Test {
+    ILido public lidoContract;
+    ILidoLocator public lidoLocator;
+    IACL public acl;
+    SecondOpinionOracle__Mock public secondOpinionOracleMock;
+    IKernel private dao;
+
+    address private rootAccount;
+    address private userAccount;
+
+    address public kernelBase;
+    address public aclBase;
+    address public evmScriptRegistryFactory;
+    address public daoFactoryAdr;
+
+    uint256 public genesisTimestamp = 1_695_902_400;
+    address private depositContract = address(0x4242424242424242424242424242424242424242);
+    address public lidoTreasury = makeAddr("dummy-lido:treasury");
+
+    LimitsList public limitList =
+        LimitsList({
+            exitedValidatorsPerDayLimit: 9000,
+            appearedValidatorsPerDayLimit: 43200,
+            annualBalanceIncreaseBPLimit: 10_00,
+            maxValidatorExitRequestsPerReport: 600,
+            maxItemsPerExtraDataTransaction: 8,
+            maxNodeOperatorsPerExtraDataItem: 24,
+            requestTimestampMargin: 7680,
+            maxPositiveTokenRebase: 750000,
+            initialSlashingAmountPWei: 1000,
+            inactivityPenaltiesAmountPWei: 101,
+            clBalanceOraclesErrorUpperBPLimit: 50
+        });
+
+    function setUpProtocol(uint256 _startBalance, address _rootAccount, address _userAccount) public {
```

Comment:
It seems that we can divide this into separate deployments since we now have two tests that might interfere with the setup. Perhaps we don't need a universal deployment at all? Instead just a bunch of deployment actions to call from test's setUp function? This is something we need to consider.